src/App.tsx

```diff
@@ -1,61 +1,133 @@
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [errorTitle, setErrorTitle] = useState(false);
+  const [errorUser, setErrorUser] = useState(false);
+
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setErrorTitle(false);
+  };
+
+  const handleChangeUser = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(+event.target.value);
+    setErrorUser(false);
+  };
+
+  const addToDo = () => {
+    if (title === '') {
+      setErrorTitle(true);
+    }
+
+    if (!selectedUser) {
+      setErrorUser(true);
+    }
+
+    if (title && selectedUser) {
+      todos.push({
+        id: Math.max(...todos.map(todo => todo.id)) + 1,
+        title: title.trim(),
+        userId: selectedUser,
+        completed: false,
+        user: getUser(selectedUser),
+      });
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    addToDo();
+
+    setTitle('');
+    setSelectedUser(0);
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleChangeTitle}
+            />
+          </label>
+
+          {errorTitle && <span className="error">Please enter a title</span>}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
-          </select>
 
-          <span className="error">Please choose a user</span>
+          <label>
+            User:
+            <select
+              data-cy="userSelect"
+              value={selectedUser}
+              onChange={handleChangeUser}
+            >
+              <option
+                value=""
+                disabled
+              >
```

Comment:
did you deploy last changes?

after changes `choose a user` option should be disabled
<img width="236" alt="image" src="https://user-images.githubusercontent.com/102411130/212004855-efbb4ac3-6a55-435a-b131-8261a8c078fe.png">